test/web3c.js

```diff
@@ -54,6 +48,33 @@ describe('Web3', () => {
     assert.notEqual(firstKey, secondKey);
   }).timeout(TIMEOUT);
 
+  it('should retrieve contract keys from a previously deployed contract address', async function() {
+    let _web3c = new web3c(gw);
+    let counterContract = new _web3c.confidential.Contract(artifact.abi);
+    try {
+      let contract = await counterContract.deploy({
+        data: artifact.bytecode
+      }).send({
+        from: address,
+        gasPrice: '0x3b9aca00'
+      });
+      let key = await _web3c.confidential.getPublicKey(contract.options.address);
+      assert.equal(64 + 2, key.public_key.length);
+    } catch (e) {
+      assert.fail(e);
+    }
+  }).timeout(TIMEOUT);
+
+  it('should not retrieve contract keys from a non deployed contract address', async function() {
+    let inst = new web3c(gw);
+    try {
+      await inst.confidential.getPublicKey('0x0000000000000000000000000000000000000000');
+      assert.equal(true, false);
+    } catch (err) {
+      // success
```

Comment:
instead of a try catch block, you can use [assert.throws](https://nodejs.org/api/assert.html#assert_assert_throws_fn_error_message)

Reply:
Good idea. Looks like we have to use [assert rejects](https://nodejs.org/api/assert.html#assert_assert_rejects_asyncfn_error_message) since it's async.